pages/listings.js

```diff
@@ -43,6 +44,8 @@ const Listings = ({ t }) => {
 Listings.getInitialProps = async ({ reduxStore }) => {
   await reduxStore.dispatch(getAlgoFirms());
   await reduxStore.dispatch(getAlgoOfferings());
+  const shuffledfirm = await reduxStore.getState().listings.firms.hits;
+  await randomizeResult(shuffledfirm);
```

Comment:
Minor point: we probably don't need `await` here as this isn't asynchronous. Would it be slightly more efficient to leave it out? I dunno. WDYT? 🤔

Reply:
The randonmize function result takes a while to resolve and sometimes return undefined, so the await there is to ensure it returns result before the other code executes. Just not taking chances with it.